magma/circuit.py

```diff
@@ -901,6 +901,9 @@ def finalize(self, dont_instantiate: bool = False):
             setattr(inst, k, v)
         return inst
 
+    def is_when_builder(self):
```

Comment:
I think we should use a free function in `primitives/when.py`. it might result in a circular import, but prefer that over adding another method to circuit.